pkg/trace/writer/trace.go

```diff
@@ -110,7 +110,7 @@ func NewTraceWriter(
 	}
 	climit := cfg.TraceWriter.ConnectionLimit
 	if climit == 0 {
-		climit = 100
+		climit = 5
```

Comment:
just an idea here: would it be easier to test this by just changing the config in staging to start and seeing how that affects performance there? If not then this is fine!